entry_types/scrolled/spec/controllers/pageflow_scrolled/editor/content_elements_controller_spec.rb

```diff
@@ -0,0 +1,131 @@
+
+        expect(response.status).to eq(401)
+      end
+
+      it 'succeeds for authorized user' do
+        section = create(:section)
+        entry = section.chapter.entry
+
+        authorize_for_editor_controller(entry)
+        post(:create,
+             params: {
+               entry_id: entry.id,
+               section_id: section.id,
+               content_element: attributes_for(:content_element, :text_block)
+             }, format: 'json')
+
+        expect(response.status).to eq(201)
+      end
+
+      it 'allows setting the content elements configuration hash' do
+        section = create(:section)
+        entry = section.chapter.entry
+
+        authorize_for_editor_controller(entry)
+        post(:create,
+             params: {
+               entry_id: entry.id,
+               section_id: section.id,
+               content_element: {
+                 configuration: {children: 'some content'}
+               }
+             }, format: 'json')
+
+        expect(section.content_elements.first.configuration).to eq('children' => 'some content')
+      end
+
+      it 'allows setting the content elements type name' do
+        section = create(:section)
+        entry = section.chapter.entry
+
+        authorize_for_editor_controller(entry)
+        post(:create,
+             params: {
+               entry_id: entry.id,
+               section_id: section.id,
+               content_element: {
+                 type_name: 'textBlock'
+               }
+             }, format: 'json')
+
+        expect(section.content_elements.first.type_name).to eq('textBlock')
+      end
+    end
+
+    describe '#update' do
+      it 'allows updating the content elements configuration hash' do
+        content_element = create(:content_element, :text_block)
+        entry = content_element.section.chapter.entry
+
+        authorize_for_editor_controller(entry)
+        patch(:update,
+              params: {
+                entry_id: entry.id,
+                id: content_element,
+                content_element: {
+                  configuration: {children: 'some content'}
+                }
+              }, format: 'json')
+
+        expect(content_element.reload.configuration).to eq('children' => 'some content')
+      end
+    end
+
+    describe '#order' do
+      it 'updates position of content elements according to given params order' do
+        section = create(:section)
+        entry = section.chapter.entry
+        content_elements = create_list(:content_element, 2, :text_block, section: section)
+
+        authorize_for_editor_controller(entry)
+
+        put(:order,
+            params: {
+              entry_id: entry.id,
+              section_id: section.id,
+              ids: [content_elements.first.id, content_elements.last.id]
+            }, format: 'json')
+
+        expect(content_elements.first.reload.position).to eq(0)
+        expect(content_elements.last.reload.position).to eq(1)
+      end
```

Comment:
Given that authorization is one of the main controller concerns, I find it a good rule of thumb to always test controllers along the line of what is permitted and what is forbidden: What are the minimal permissions required to perform an action successfully. What are the maximal permissions not yet sufficient to perform it. For simple actions that can be authenticated vs not authenticated or authorized vs not authorized. For actions with more complex params (like this one), I try to come up with cases that combine representative inputs from different classes, e.g. permitted content element ids but section id of other entry etc.